File: storable-data/src/main/java/dev/m4tt3o/storable/data/entity/FileNode.java
package dev.m4tt3o.storable.data.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "nodes")
public class FileNode {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "owner_id", nullable = false)
    private String ownerId;

    @Column(name = "parent_id")
    private Long parentId;

    @Column(nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private NodeKind kind;

    private Long size;
    private String mime;

    @Column(name = "storage_key")
    private String storageKey;

    @Column(name = "created_at", insertable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "modified_at", insertable = false, updatable = false)
    private LocalDateTime modifiedAt;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    public enum NodeKind {
        file,
        folder
    }
}

File: web/types/FileNode.ts
export interface FileNode {
  id: number;
  name: string;
  size: number | null;
  mime: string | null;
  storageKey: string | null;
  createdAt: string;
  modifiedAt: string;
  isDeleted: boolean;
  deletedAt: string | null;
  ownerId: string;
  parentId: number | null;
  folder: boolean;
}
